tests/components/sensor/test_yr.py

```diff
@@ -1,77 +1,76 @@
-                                'sensor': {'platform': 'yr',
-                                           'elevation': 0}})
-
-        state = self.hass.states.get('sensor.yr_symbol')
-
-        assert '3' == state.state
-        assert state.state.isnumeric()
-        assert state.attributes.get('unit_of_measurement') is None
-
-    def test_custom_setup(self, requests_mock):
-        """Test a custom setup."""
-        requests_mock.get('http://api.yr.no/weatherapi/locationforecast/1.9/',
-                          text=load_fixture('yr.no.json'))
-        now = datetime(2016, 6, 9, 1, tzinfo=dt_util.UTC)
+def _setup_platform(hass, config):
+    """Setup a yr platform."""
+    hass.allow_pool = True
 
+    def _setup():
+        """Setup the platform."""
         with patch('homeassistant.components.sensor.yr.dt_util.utcnow',
-                   return_value=now):
-            assert setup_component(self.hass, 'sensor', {
-                                    'sensor': {'platform': 'yr',
-                                               'elevation': 0,
-                                               'monitored_conditions': [
-                                                   'pressure',
-                                                   'windDirection',
-                                                   'humidity',
-                                                   'fog',
-                                                   'windSpeed']}})
-
-        state = self.hass.states.get('sensor.yr_pressure')
-        assert 'hPa' == state.attributes.get('unit_of_measurement')
-        assert '1009.3' == state.state
-
-        state = self.hass.states.get('sensor.yr_wind_direction')
-        assert '°' == state.attributes.get('unit_of_measurement')
-        assert '103.6' == state.state
-
-        state = self.hass.states.get('sensor.yr_humidity')
-        assert '%' == state.attributes.get('unit_of_measurement')
-        assert '55.5' == state.state
-
-        state = self.hass.states.get('sensor.yr_fog')
-        assert '%' == state.attributes.get('unit_of_measurement')
-        assert '0.0' == state.state
-
-        state = self.hass.states.get('sensor.yr_wind_speed')
-        assert 'm/s', state.attributes.get('unit_of_measurement')
-        assert '3.5' == state.state
+                   return_value=NOW), assert_setup_component(1):
+            setup_component(hass, 'sensor', {'sensor': config})
+    return _setup
+
+
+@asyncio.coroutine
+def test_default_setup(hass, aioclient_mock):
+    """Test the default setup."""
+    aioclient_mock.get('http://api.yr.no/weatherapi/locationforecast/1.9/',
+                       text=load_fixture('yr.no.json'))
+    config = {'platform': 'yr',
+              'elevation': 0}
+    yield from hass.loop.run_in_executor(None, _setup_platform(hass, config))
+
+    state = hass.states.get('sensor.yr_symbol')
+
+    assert state.state == '3'
+    assert state.state.isnumeric()
```

Comment:
This assert makes no sense, we know the state is `'3'` so we know it will be numeric.